### PR TITLE
Use versions from `grails-bom` in example apps

### PR DESCRIPTION
**Pull request by matrei**
And remove unused versions from `gradle.properties`

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,9 +1,6 @@
 projectVersion=9.0.0-SNAPSHOT
 #4.0.2-4.0.3 breaks org.ysb33r.gradle:grolifant for chromedriver
 asciidoctorGradleVersion=4.0.1
```
Comment (jdaugherty) [inline gradle.properties:2]: Grails-data-mapping & gorm-hibernate5 are meant to be used outside of Grails applications.  Are any of these versions still required to set version properties correctly when grails-bom isn't in use?  

I'm assuming the jakarta version due to the older hibernate import may be the only one?  The rest look like tests or grails specific ones.

File: examples/grails3-hibernate5/build.gradle [hunk 2]
```diff
@@ -5,33 +5,33 @@ dependencies {
     implementation "org.springframework.boot:spring-boot-autoconfigure"
     implementation "org.springframework.boot:spring-boot-starter-actuator"
     implementation "org.springframework.boot:spring-boot-starter-tomcat"
```
Comment (jdaugherty) [inline examples/grails3-hibernate5/build.gradle]: Are the other implementation statements even needed with the bom?

File: examples/grails3-hibernate5/build.gradle [hunk 3]
```diff
@@ -5,33 +5,33 @@ dependencies {
     implementation "org.springframework.boot:spring-boot-autoconfigure"
```
Comment (jdaugherty) [inline examples/grails3-hibernate5/build.gradle]: I'm guessing the grails gradle plugin includes the grails-bom by default?  I'm not as familiar with the way the bom's work - does it just pull in the versions or does it automatically add the dependencies too?  Are any of these implementations even needed if the bom is included?

**Review by jdaugherty — APPROVED** [review]

File: examples/grails3-hibernate5/build.gradle (hunk 3 above)
Comment (matrei) [reply examples/grails3-hibernate5/build.gradle]: Yes, the `grails-gradle-plugin` adds the `grails-bom` to the project.

A `bom` only provides suggested versions for the modules it has listed, so you don't have to provide specific versions yourself. It does not add any dependencies by itself.

File: examples/grails3-hibernate5/build.gradle (hunk 2 above)
Comment (matrei) [reply examples/grails3-hibernate5/build.gradle]: A `bom` will not change the need to import all required dependencies.

I have not cleaned up the list of dependencies in this PR. Only removed the versions that are already provided by `grails-bom`. I am sure the dependency list can be cleaned up a bit as well.

File: gradle.properties (hunk 1 above)
Comment (matrei) [reply gradle.properties:2]: Yes, all remaining versions in `gradle.properties` are still used somewhere in the project.

This PR (so far) only aims to clean up versions from the test/example apps by using the versions listed in the `grails-bom`.
(I'm sure we can cleanup the project even further down the line.)

**Review by jamesfredley — APPROVED** [review]

**Review by codeconsole — CHANGES_REQUESTED** [review]
@matrei have you tried running all these examples to see if they work as expected?

Do you know why `@EnableAutoConfiguration(exclude = DataSourceTransactionManagerAutoConfiguration)` was there and commented out and if it still needs to be excluded.  It can be excluded in application.yml if necessary.

**Comment by matrei** [comment]
>  have you tried running all these examples to see if they work as expected?

Yes

> Do you know why @EnableAutoConfiguration(exclude = DataSourceTransactionManagerAutoConfiguration) was there and commented out and if it still needs to be excluded. It can be excluded in application.yml if necessary.

This was probably to prevent Spring Boot from configuring a transaction manager as GORM also does this and there is a problem when two are configured. It seems this is not needed anymore when using the `org.grails.plugins:hibernate5` plugin as it sets it's transaction manager as primary.

I actually had to add something like that to `examples-spring-boot-hibernate5` to make it not bomb when accessing the database from the started app: `@SpringBootApplication(exclude = HibernateJpaAutoConfiguration)`. It seems, this is not handled by default by the `gorm-hibernate5-spring-boot` plugin.

**Comment by codeconsole** [comment]
> I actually had to add something like that to `examples-spring-boot-hibernate5` to make it not bomb when accessing the database from the started app: `@SpringBootApplication(exclude = HibernateJpaAutoConfiguration)`. It seems, this is not handled by default by the `gorm-hibernate5-spring-boot` plugin.

The correct behavior I am finding is to not have to use these at all and to reconfigure the plugins to work with the AutoConfigurations.  In most cases the AutoConfigurations do 2x the work because what happens is the AutoConfigurations run first, create the service and then Grails creates the same service with the same name.   This is terribly inefficient. 

The best fixes in these scenarios are to create the bean definition before the Autoconfiguration. When not possible, you can also create a javaconfig for a null placeholder definition so that it stops the AutoConfiguration as well.

These issues are usually masked due to bean [definition overriding.](https://github.com/grails/grails-core/issues/13827) enabled by default

**Review by codeconsole — APPROVED** [review]

**Comment by codeconsole** [comment]
@matrei let's just merge this because it is broken without it.

**Comment by codeconsole** [comment]
@matrei ?

**Comment by matrei** [comment]
Hi 😄 ! I have and was going to to some more work in this PR. But I seem to get stuck elsewhere.

**Pull request merged**